### PR TITLE
Replace JS script with Dart.

### PR DESCRIPTION
**Pull request by isoos**
Mostly a line-by-line rewrite, except for the `scrollTo` function. I've tried to get it working with `dartdevc`, but couldn't figure out a way to get the compiled (and short) script running as the compiled version.

Backstory: I wanted to add more robust history and hash handling (e.g. clicking on a `#relative-link` doesn't change the URL, nor adds it to history), but JS is too brittle for that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,12 @@ Dart support for AppEngine).
 
 ## Local Development
 
+### Compile script.dart to script.dart.js
+
+```
+dart2js static/v2/js/script.dart -o static/v2/js/script.dart.js
```
Comment (kevmoo) [inline README.md]: Run with `--minify --trust-primitives --trust-type-annotations`

From 310054 --> 106947

Comment (isoos) [reply README.md]: Bummer, I had been running it with `-m` but removed it for DDC experiments.

File: app/test/frontend/golden/v2/web_landing_page.html [hunk 2]
```diff
@@ -111,6 +111,5 @@ <h3 class="title"><a href="/experimental/packages/foobar_pkg">foobar_pkg</a></h3
   <a class="link" href="https://www.google.com/intl/en/policies/privacy/">Privacy Policy</a>
 </footer>
 
-<script src="/static/v2/js/script.js" defer></script>
```
Comment (kevmoo) [inline app/test/frontend/golden/v2/web_landing_page.html]: Wow! we had this twice?

**Comment by isoos** [comment]
> Annoyed that the code is so big.

+1. Maybe it could be a benchmark goal for `dart2js`?

**Pull request merged**